runtime/src/main/java/org/corfudb/runtime/view/stream/IStreamView.java

```diff
@@ -0,0 +1,123 @@
+package org.corfudb.runtime.view.stream;
+
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+
+/** This interface represents a view on a stream. A stream is an ordered
+ * set of log entries which can only be appended to and read in sequential
+ * order.
+ *
+ * Created by mwei on 1/5/17.
+ */
+public interface IStreamView extends Iterator<LogData> {
+
+    /** Return the ID of the stream this view is for.
+     * @return  The ID of the stream.
+     */
+    UUID getID();
+
+    /** Reset the state of this stream view, causing the next read to
+     * start from the beginning of this stream.
+     */
+    void reset();
+
+    /** Append an object to the stream, returning the global address
+     * it was written at.
+     * <p>
+     * Optionally, provide a method to be called when an address is acquired,
```

Comment:
explain better: 
Optionally, provide a method to be called when an address is acquired for this append, but has not been written yet. 

This is necessary for operations like compare-and-swap (CAS). We need to sync state, up to the acquired address, before applying CAS.